pelita/layout.py

```diff
@@ -74,13 +75,13 @@ def get_available_layouts(size='normal', dead_ends=False):
         size = ''
 
     av_layouts = []
-    for resource in importlib_resources.files('pelita._layouts').iterdir():
-        if resource.is_file() and resource.name.endswith('.layout') and size in resource.name:
-            layout_name = resource.name.removesuffix('.layout')
-            if dead_ends and 'dead_ends' in resource.name:
-                av_layouts.append(layout_name)
-            if not dead_ends and 'dead_ends' not in resource.name:
-                av_layouts.append(layout_name)
+    for file in os.listdir(importlib_resources.files('pelita._layouts')):
```

Comment:
Very interesting. I think part of the speed gain comes from skipping the `is_file` check but even when narrowing it down with `glob` I cannot get the same speed as with `listdir`. It does lead me to believe, however, that we should do the `get_available_layouts` differently.

Reply:
I spent seom time trying to understanding it too. I think the `pathlib` is just slow. And I don't see an easy way of changing the code much, because we need `importlib.resources`. If we use the `.contents` attribute it is even faster, and it returns just a list of strings. Unfortunately that attribute is deprecated and this leaves us with only `.files`... `glob` and `iterdir` are both slower, even if `glob` is a bit better.
We could decide to hardcode the layout names and just assume that the files are going to be there. `get_available_layouts` doesn't need to be dynamic as it is now and can file at runtime if then for some absurd reasons the files are not there.